File: ProjectExchange.UI/ProjectExchange/src/app/models/Enums/operation-type.ts
export class OperationType {
  private constructor(value: string) { this.value = value; }
  value: string;
  public static Take: OperationType = new OperationType("Take");
  public static Sell: OperationType = new OperationType("Sell");
  public static Load: OperationType = new OperationType("Load");
  public static Deposit: OperationType = new OperationType("Deposit");
}

File: ProjectExchange.API/ApiGateways/ExchangeGateway/Models/EntityModels/Enums/OperationType.cs
namespace ExchangeGateway.Models.EntityModels.Enums {
    public class OperationType {

        //* Enum kullanımı gibi string kullanımı bknz:https://stackoverflow.com/questions/630803/associating-enums-with-strings-in-c-sharp?page=1&tab=votes#tab-top
        public OperationType (string value) { Value = value; }
        public string Value { get; set; }
        public static OperationType Take { get { return new OperationType (nameof (Take)); } }
        public static OperationType Sell { get { return new OperationType (nameof (Sell)); } }
        public static OperationType Load { get { return new OperationType (nameof (Load)); } }
        public static OperationType Deposit { get { return new OperationType (nameof (Deposit)); } }
    }
}